cpp/src/gandiva/jni/jni_common.cc

```diff
@@ -640,14 +672,11 @@ Java_org_apache_arrow_gandiva_evaluator_JniWrapper_evaluateProjector(
           std::make_shared<arrow::MutableBuffer>(value_buf, data_sz);
 
       auto array_data =
-          arrow::ArrayData::Make(field->type(), num_rows, {bitmap_buf, data_buf});
+          arrow::ArrayData::Make(field->type(), actual_row_count, {bitmap_buf, data_buf});
       output.push_back(array_data);
     }
-    if (!status.ok()) {
-      break;
-    }
-
-    status = holder->projector()->Evaluate(*in_batch, output);
+    status =
+        holder->projector()->Evaluate(*in_batch, sel_vec_type, *selection_vector, output);
```

Comment:
the selection_vector can be null - the Evaluate API should take a pointer instead of ref.